zulip_bots/zulip_bots/bots/chess/chess.py

```diff
@@ -0,0 +1,703 @@
+import chess
+import chess.uci
+import re
+import copy
+
+class ChessHandler(object):
+    def usage(self):
+        return (
+            'Chess Bot is a bot that allows you to play chess against either '
+            'another user or the computer. Use `start with other user` or '
+            '`start as <color> with computer` to start a game.\n\n'
+            'In order to play against a computer, `chess.conf` must be set '
+            'with the key `stockfish_location` set to the location of the '
+            'Stockfish program on this computer.'
+        )
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('chess')
+
+        try:
+            self.engine = chess.uci.popen_engine(
+                self.config_info['stockfish_location']
+            )
+            self.engine.uci()
+        except FileNotFoundError:
+            # It is helpful to allow for fake Stockfish locations if the bot
+            # runner is testing or knows they won't be using an engine.
+            print('That Stockfish doesn\'t exist. Continuing.')
+
+    def handle_message(self, message, bot_handler):
+        START_REGEX = 'start with other user$'
+        START_COMPUTER_REGEX = (
+            'start as (?P<user_color>white|black) with computer'
```

Comment:
Not that it is that important, but we should avoid compiling regex in loops. (Might become more important later on when we decide to run bots directly on the chat server)

Reply:
@tommyip OK. How should I go about compiling the RegEx outside of the loop? Thanks!

Reply:
Just re.compile it in the global namespace.

Reply:
By the loop I just mean the handle_message function, since that is called on everything message it receives.

Reply:
@tommyip Oh, I understand now! 😃 Thanks! I fixed that in commit 4a31674.